client/client_linux.go

```diff
@@ -286,3 +286,78 @@ func GetExtendedReportAtVmpl(d Device, userData [64]byte, vmpl int) (*pb.Attesta
 func GetExtendedReport(d Device, userData [64]byte) (*pb.Attestation, error) {
 	return GetExtendedReportAtVmpl(d, userData, 0)
 }
+
+// GuestFieldSelect represents which guest-provided information will be mixed into a derived key.
+type GuestFieldSelect struct {
+	TCBVersion  bool
+	GuestSVN    bool
+	Measurement bool
+	FamilyID    bool
+	ImageID     bool
+	GuestPolicy bool
+}
+
+// SnpDerivedKeyReq represents a request to the SEV guest device to derive a key from specified
+// information.
+type SnpDerivedKeyReq struct {
+	// UseVCEK determines of the derived key will be based on VCEK or VMRK. This is opposite from the
+	// ABI's ROOT_KEY_SELECT to avoid accidentally making an unsafe choice in a multitenant
+	// environment.
+	UseVCEK          bool
```

Comment:
golang allows default values for structs -- right? If yes, should we make this default to `true`?

Reply:
I specifically made this UseVCEK to make its false default value choose VMRK. The VCEK is more dangerous to use than VMRK, so I wanted users who want that behavior to explicitly request it.

Reply:
Got it now. Thank you for explaining. After reading the new LIMITATIONS text, this is now very clear to me.